bundle/bundle-writer.go

```diff
@@ -0,0 +1,352 @@
+    Version          string `json:"version"`
+    RepositoryURL    string `json:"repository_url"`
+	ValuesFile       string `json:"values_file"`
+}
+
+type BundleConfiguration struct {
+    Repo   string `json:"repo"`
+    Branch string `json:"branch"`
+}
+
+type BundleBaseTemplates struct {
+    Repo     string `json:"repo"`
+    Branch   string `json:"branch"`
+    Stencils []*BundleStencil `json:"stencils"`
+}
+
+type Metadata struct {
+    App         string     `json:"app"`
+    Timestamp   time.Time  `json:"timestamp"`
+    Annotations []string   `json:"annotations"`
+}
+
+type BundleStencil struct {
+    Uid              string   `json:"uid"`
+    Filename         string   `json:"filename"`
+    TemplateFilename string   `json:"template_filename"`
+    ContextID        string   `json:"context_id"`
+    Status           int      `json:"status"`
+    Tags             []string `json:"tags"`
+    Sequence         int      `json:"sequence"`
+}
+
+type BundleStencilGroup struct {
+    Uid  string   `json:"uid"`
+    Name string   `json:"name"`
+    Tags []string `json:"tags"`
+}
+
+type BundlePolicy struct {
+    Uid      string   `json:"uid"`
+    Name     string   `json:"name"`
+    Selector string   `json:"selector"`
+    Tags     []string `json:"tags"`
+}
+
+
+func CreateSkycapFiles(outputDir string,
+						templateRepository string,
+						branch string,
+						pack_name string,
+						githubURL string,
+						services []*common.Service,
+						databases []common.Database) error {
+
+	if templateRepository == "" {
+		//no stencil template defined for this pack, print an error and do nothing
+		fmt.Printf("Sorry but there is no stencil template for this language/framework yet\n")
+		return nil
+	}
+	//Create .bundle directory structure if it doesn't exist
+	skycapFolder := filepath.Join(outputDir, "skycap")
+	err := createBundleFolderStructure(skycapFolder)
+	defer os.RemoveAll(filepath.Join(skycapFolder, "temp"))
+
+	//create manifest.json file and start filling
+	manifestFile, err := loadManifest()
+	if err != nil {
+		return err
+	}
+
+	manifestFile, err = saveEnvVars(pack_name, getEnvVars(services, databases), manifestFile, skycapFolder)
+	if err != nil {
+		return err
+	}
+
+	manifestFile, err = addDatabase(manifestFile, databases)
+
+	manifestFile, err = getRequiredStencils(
+		templateRepository,
+		branch,
+		outputDir,
+		services,
+		skycapFolder,
+		manifestFile,
+		githubURL)
+
+	if err != nil {
+		return err
+	}
+
+	manifestFile, err = addMetadatas(manifestFile)
+
+	if err != nil {
+		return err
+	}
+
+	err = saveManifest(skycapFolder, manifestFile)
+	if err !=nil {
+		return err
+	}
+	return err
+}
+
+// downloading templates from github and putting them into homedir
+func getStencilTemplateFile(templateRepository string, skycapFolder string, filename string, branch string) (string, error) {
+
+	//Download templates.json file
+	manifest_path := templateRepository + filename // don't need to use filepath since it's a URL
```

Comment:
always use `path.Join` to make sure your code works on all OSs

Reply:
In this case, manifest_path is the URL that will be used to download the file (filename) from the templateRepository(BTR), so we need the '/' in any case. I can rename the `manifest_path` variable for it to be more clear.